From 1b6d40a8da5a1d5d2249f2f7d032e3ece041238c Mon Sep 17 00:00:00 2001
From: Ahmad Samir <a.samirh78@gmail.com>
Date: Thu, 2 Oct 2025 18:07:31 +0300
Subject: [PATCH 316/553] QObject: take by QSpan in static helper function

Instead of passing by a pointer and length.

Change-Id: I591f67a1af0ef10af9456adc009f21eb2cc6a1a3
Reviewed-by: Thiago Macieira <thiago.macieira@intel.com>
(cherry picked from commit 6f5cdac6b6fec6b217629b0bf3e217e81384f39c)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/kernel/qobject.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/corelib/kernel/qobject.cpp b/src/corelib/kernel/qobject.cpp
index 469994aa3c7..4c5d9b83a1b 100644
--- a/src/corelib/kernel/qobject.cpp
+++ b/src/corelib/kernel/qobject.cpp
@@ -104,8 +104,9 @@ static int *queuedConnectionTypes(const QMetaMethod &method)
 }
 
 // ### Future work: replace with an array of QMetaType or QtPrivate::QMetaTypeInterface *
-static int *queuedConnectionTypes(const QArgumentType *argumentTypes, int argc)
+static int *queuedConnectionTypes(QSpan<const QArgumentType> argumentTypes)
 {
+    const int argc = int(argumentTypes.size());
     auto types = std::make_unique<int[]>(argc + 1);
     for (int i = 0; i < argc; ++i) {
         const QArgumentType &type = argumentTypes[i];
@@ -3072,8 +3073,7 @@ QMetaObject::Connection QObject::connect(const QObject *sender, const char *sign
     // ### Future work: attempt get the metatypes from the meta object first
     // because it's possible they're all registered.
     int *types = nullptr;
-    if ((type == Qt::QueuedConnection)
-            && !(types = queuedConnectionTypes(signalTypes.constData(), signalTypes.size()))) {
+    if (type == Qt::QueuedConnection && !(types = queuedConnectionTypes(signalTypes))) {
         return QMetaObject::Connection(nullptr);
     }
 
-- 
2.51.2

